### PR TITLE
feat: services section

### PR DESCRIPTION
**Pull request by RanBerkovich1**
I created the services section with media queries, I used the ones from the [wix website ](https://codux-1711972748.wixstudio.io/my-site-3/blank-1?rc=test-site)

https://github.com/user-attachments/assets/9db32e18-7d48-4d60-8bf2-fb12c1a0676c

### CODE REVIEW

File: _codux/boards/services-section.board.module.scss [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    background-repeat: repeat;
+    background-attachment: scroll;
+    background-origin: padding-box;
+    background-clip: border-box;
+    gap: 60px;
+    flex-direction: row;
+}
+
+.servicesTitle {
+    color: 000000;
```
Comment (alisey) [inline _codux/boards/services-section.board.module.scss]: Not a valid color.

File: _codux/boards/services-section.board.module.scss [hunk 2]
```diff
@@ -0,0 +1,88 @@
+.servicesSection {
+    display: flex;
+    padding: 76px;
+    background-image: linear-gradient(90deg, #e1e2ff 0%, #e3fbf9 100%);
+    background-position: 0%;
+    background-size: auto;
+    background-repeat: repeat;
+    background-attachment: scroll;
+    background-origin: padding-box;
+    background-clip: border-box;
```
Comment (alisey) [inline _codux/boards/services-section.board.module.scss]: I guess this was added by Codux. If it's not necessary let's remove it.

File: _codux/boards/services-section.board.module.scss [hunk 3]
```diff
@@ -0,0 +1,88 @@
+.servicesSection {
```
Comment (alisey) [inline _codux/boards/services-section.board.module.scss]: If I use this snippet for anything other than a list of services (e.g. a features section, testimonials section), then these class names will not make sense.

So if you have a better idea how to name this layout, let's do that.

Also I see that you've used both prefixed and non-prefixed names:
* Prefixed: "servicesSection", "servicesTitle", "servicesDescription", "serviceCardTitle"
* Non-prefixed: "card", "content", "icon".

Both approaches have pros and cons, but we should pick one convention and be consistent.

File: _codux/boards/services-section.board.module.scss
Comment (alisey) [inline _codux/boards/services-section.board.module.scss:1]: Run `prettier` on this file.

File: _codux/boards/hero-flex.board.tsx [hunk 4]
```diff
@@ -28,6 +28,6 @@ export default createBoard({
         </div>
     ),
     environmentProps: {
-        windowWidth: 430,
+        windowWidth: 360,
```
Comment (alisey) [inline _codux/boards/hero-flex.board.tsx]: Looks like a change that doesn't belong in this PR.

File: _codux/boards/services-section.board.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    This is the space to introduce visitors to the business or brand. Briefly
+                    explain who&apos;s behind it, what it does and what makes it unique. Share its
+                    core values and what this site has to offer.
+                </p>
+                <button className={commonStylesStyles.secondaryButtonWidth}>Explore</button>
+            </div>
+            <div className={styles.allCards}>
+                <div className={styles.card}>
+                    <div className={styles.iconContainer}>
+                        <img src={AnchorSvg} alt="" className={styles.icon} />
```
Comment (alisey) [inline _codux/boards/services-section.board.tsx]: `AnchorSvg` is a variable, and Add Panel snippets don't support variables. I know it sucks, but you'll have to upload these images to wix media platform or something, and use ``

File: _codux/boards/services-section.board.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    Board: () => (
+        <div className={styles.servicesSection}>
+            <div className={styles.content}>
+                <h2 className={styles.servicesTitle}> Write a Title Here</h2>
+                <p className={styles.servicesDescription}>
+                    This is the space to introduce visitors to the business or brand. Briefly
+                    explain who&apos;s behind it, what it does and what makes it unique. Share its
+                    core values and what this site has to offer.
+                </p>
+                <button className={commonStylesStyles.secondaryButtonWidth}>Explore</button>
```
Comment (alisey) [inline _codux/boards/services-section.board.tsx]: @ynesteruk do our built-in board snippets support multiple CSS files in a single snippet?

@RanBerkovich1 in any case don't name it `commonStylesStyles`. And `.secondaryButtonWidth` doesn't make sense as a common class.

File: _codux/boards/services-section.board.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                        clients can benefit from it. It’s an opportunity to add a short description
+                        that includes relevant details.
+                    </p>
+                </div>
+                <div className={styles.card}>
+                    <div className={styles.iconContainer}>
+                        <img src={LeafSvg} alt="" className={styles.icon} />
+                    </div>
+                    <h6 className={styles.serviceCardTitle}>Service Title</h6>
+                    <p className={styles.serviceCardDescription}>
```
Comment (alisey) [inline _codux/boards/services-section.board.tsx]: Using `<p>` elements in visual tools like Codux is problematic.

If a user drops a `<div>` or `<hr>` into a `<p>` element, it will split it into two `<p>` elements, as shown here: https://jsfiddle.net/1og7rsqm/

This behavior follows the HTML spec, but it can be confusing to our users. I suggest we avoid using `<p>`.

File: _codux/boards/services-section.board.tsx (hunk 6 above)
Comment (alisey) [reply _codux/boards/services-section.board.tsx]: Also I see this project imports `src/styles/index.scss` into every board via `board-global-setup`. Any styles coming from `index.css` will not be available in the snippet.

E.g. the this font will not be used:
https://github.com/wixplosives/wireframe-snippets/blob/32b5f322a0f7aae58c3a0c1331c60a9ec737f188/src/styles/index.scss#L5-L14

I don't think you can currently add a custom font to the snippet. You'll either have to omit the font entirely, and then it will depend on the font used in each specific user's project. Or you can hardcode a built-in font like Helvetica, but then users will need to replace it each time after adding the snippet.

Comment (ynesteruk) [reply _codux/boards/services-section.board.tsx]: > do our built-in board snippets support multiple CSS files in a single snippet?

No. at the moment it should be only one plain css file that located in the same folder and named according to our convention (e.g. `example.board.css` belongs to `example.board.tsx`)

Comment (alisey) [reply _codux/boards/services-section.board.tsx]: @ynesteruk so no scss files, no css modules, no scoping?

Comment (ynesteruk) [reply _codux/boards/services-section.board.tsx]: @alisey yes. We made a basic solution to support simple cases (e.g. grid layout)

File: _codux/boards/hero-flex.board.tsx (hunk 4 above)
Comment (RanBerkovich1) [reply _codux/boards/hero-flex.board.tsx]: to remove it from here, this is the right thing to do? or just put back the 430 
thanks 

image

File: _codux/boards/services-section.board.tsx (hunk 7 above)
Comment (RanBerkovich1) [reply _codux/boards/services-section.board.tsx]: So, should I use div instead?

File: _codux/boards/hero-flex.board.tsx (hunk 4 above)
Comment (alisey) [reply _codux/boards/hero-flex.board.tsx]: I think if you click "Delete file" it will actually delete the file, not just the change. It's easier to change the number back, or if the new number is OK then just keep it.

I just wasn't sure if it's an accidental change or not.

File: _codux/boards/services-section.board.tsx (hunk 7 above)
Comment (alisey) [reply _codux/boards/services-section.board.tsx]: Up to you. Depends on whether you think the issue I described is important.

File: src/assets/anchor.svg
Comment (alisey) [inline src/assets/anchor.svg:1]: Remove these SVG files.

File: _codux/boards/services-section.board.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
+import { createBoard } from '@wixc3/react-board';
+import classNames from 'classnames';
+import styles from './services-section.board.module.scss';
+
+export default createBoard({
+    name: 'ServicesSection',
+    Board: () => (
+        <div className={styles.servicesSection}>
```
Comment (alisey) [inline _codux/boards/services-section.board.tsx]: As we discussed in the daily, we'll have to use `<div className="servicesSection">` instead of `<div className={styles.servicesSection}>`

Also because we realized we can't use CSS/SASS modules, now it's more important that all classes in the snippet have a common unique prefix. That means:

* `servicesSection`
* `servicesSectionButton`
* `servicesSectionCards`
* `servicesSectionIcon`
* `servicesSectionIconContainer`
* `servicesSectionTitle`
* and so on

File: _codux/boards/services-section.board.module.scss [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    width: fit-content;
+    display: flex;
+    align-items: center;
+    cursor: pointer;
+    padding-right: 30px;
+    padding-left: 30px;
+    border: 1px solid #000000;
+    border-radius: 50px;
+    &:hover {
+    }
```
Comment (alisey) [inline _codux/boards/services-section.board.module.scss]: Remove this one.

File: _codux/boards/services-section.board.module.scss [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    align-content: center;
+    display: flex;
+    justify-content: end;
+    padding-bottom: 24px;
+}
+.icon {
+    width: 32px;
+}
+
+@media (width <= 1000px) {
```
Comment (alisey) [inline _codux/boards/services-section.board.module.scss]: Move all media queries to the bottom of the file.

File: _codux/boards/hero-flex.board.tsx (hunk 4 above)
Comment (RanBerkovich1) [reply _codux/boards/hero-flex.board.tsx]: probably just played with it - so accidental. I'll change it back, thanks.

File: _codux/boards/services-section.board.tsx (hunk 8 above)
Comment (ynesteruk) [reply _codux/boards/services-section.board.tsx]: > Also because we realized we can't use CSS/SASS modules, now it's more important that all classes in the snippet have a common unique prefix.

I'm not sure we need it. When we drop a snippet into a component with a CSS/SCSS modules, it doesn't matter. 
And in case of a plain CSS/SCSS a prefix will be autogenerated (it is relevant for built-in snippet, but I guess we will reuse it)

Comment (alisey) [reply _codux/boards/services-section.board.tsx]: How the prefix is generated? Is it based on the board file name / board "name" property?

Comment (ynesteruk) [reply _codux/boards/services-section.board.tsx]: It is a board file name or a component name

Comment (alisey) [reply _codux/boards/services-section.board.tsx]: > When we drop a snippet into a component with a CSS/SCSS modules, it doesn't matter

If I add 'services section' and 'hero banner' snippets to my component, and if both of those snippets contain a button, then in the resulting component we'll have 'button1' and 'button2'.

Comment (ynesteruk) [reply _codux/boards/services-section.board.tsx]: From this perspective prefixing makes sense

Comment (alisey) [reply _codux/boards/services-section.board.tsx]: But does it mean that if we add a prefix to each class name, then Codux will add another prefix and we'll end up with `.servicesSectionServicesSectionTitle`?

Comment (ynesteruk) [reply _codux/boards/services-section.board.tsx]: It will be something like `ComponentName_servicesSectionTitle`
If a target component is `ServicesSeciton` then it's a `ServicesSection_servicesSectionTitle` (and only for a  target component with a plain css/scss)

Comment (alisey) [reply _codux/boards/services-section.board.tsx]: To summarize:

1. If the destination component uses plain CSS:
   `<div className="button">` will be pasted as `<div className="SnippetName_button">`
2. If the destination component uses CSS modules:
   `<div className="button">` will be pasted as `<div className={styles.button}>`

Manually prefixing classes in the snippet's CSS file is bad for scenario 1, but good for scenario 2.

I think scenario 2 is more common. Our own templates use modules by default. Which means we should add manual prefixes.

File: _codux/boards/services-section.board.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+                <h2 className="servicesTitle"> Write a Title Here</h2>
+                <p className="servicesDescription">
+                    This is the space to introduce visitors to the business or brand. Briefly
+                    explain who&apos;s behind it, what it does and what makes it unique. Share its
+                    core values and what this site has to offer.
+                </p>
+                <button className="servicesButton">Button</button>
+            </div>
+            <div className="servicesCards">
+                <div className="card servicesCard">
```
Comment (alisey) [inline _codux/boards/services-section.board.tsx:18]: `className="card servicesCard"` - can't use two classes here.

File: _codux/boards/services-section.board.css [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+.servicesCard {
+  padding: 24px;
+  background-color: #ffffff;
+  border-radius: 16px;
+  box-shadow: 0.71px 0.71px 20px rgba(0, 0, 0, 0.08);
+  min-width: 265px;
+}
+
+.serviceCardDescription {
```
Comment (alisey) [inline _codux/boards/services-section.board.css:61]: Should be ".servicesCardDescription"

File: _codux/boards/services-section.board.css [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.serviceCardTitle {
```
Comment (alisey) [inline _codux/boards/services-section.board.css:64]: Should be ".servicesCardTitle"

File: _codux/boards/services-section.board.css [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+  padding-bottom: 8px;
+}
+.servicesCardIconContainer {
+  width: 100%;
+  align-content: center;
+  display: flex;
+  justify-content: end;
+  padding-bottom: 24px;
+}
+.servicesCardIcon {
```
Comment (alisey) [inline _codux/boards/services-section.board.css:77]: Separate rules in the file by empty lines. Run prettier on the file. The indentation should be 4 characters instead of 2.

File: _codux/boards/services-section.board.css [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+  grid-template-columns: 1fr 1fr;
+  gap: 20px;
+
+  @media (width <= 1024px) {
+    width: auto;
+    display: grid;
+    grid-template-rows: 1fr 1fr;
+    grid-template-columns: 1fr 1fr;
+    gap: 20px;
+  }
```
Comment (alisey) [inline _codux/boards/services-section.board.css:42]: This entirely duplicates the default style.

File: _codux/boards/services-section.board.css [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+  @media (width <= 768px) {
+    width: auto;
+    display: grid;
+    grid-template-rows: 1fr;
+    grid-template-columns: 1fr;
+    gap: 20px;
+  }
```
Comment (alisey) [inline _codux/boards/services-section.board.css:50]: Everything here apart from `grid-template-` duplicates the default style.

File: _codux/boards/services-section.board.css [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+  line-height: 1.6em;
+  padding-bottom: 28px;
+}
+.servicesContent {
+  display: flex;
+  flex-direction: column;
+  min-width: 275px;
+}
+.servicesCards {
+  width: auto;
```
Comment (alisey) [inline _codux/boards/services-section.board.css:30]: width is auto by default.

File: _codux/boards/services-section.board.css [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+    grid-template-columns: 1fr;
+    gap: 20px;
+  }
+}
+
+.servicesCard {
+  padding: 24px;
+  background-color: #ffffff;
+  border-radius: 16px;
+  box-shadow: 0.71px 0.71px 20px rgba(0, 0, 0, 0.08);
```
Comment (alisey) [inline _codux/boards/services-section.board.css:57]: `0.71px 0.71px` -> `0 0`

File: _codux/boards/services-section.board.css [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+  padding-bottom: 16px;
+}
+.servicesDescription {
+  font-size: 16px;
+  line-height: 1.6em;
+  padding-bottom: 28px;
+}
+.servicesContent {
+  display: flex;
+  flex-direction: column;
```
Comment (alisey) [inline _codux/boards/services-section.board.css:26]: This doesn't do anything.

File: _codux/boards/services-section.board.css [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.serviceCardTitle {
+  font-size: 20px;
+  line-height: 1.5em;
+  font-weight: 700;
+  padding-bottom: 8px;
+}
+.servicesCardIconContainer {
+  width: 100%;
```
Comment (alisey) [inline _codux/boards/services-section.board.css:71]: Block elements stretch to 100% by default.

File: _codux/boards/services-section.board.css [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+  align-content: center;
```
Comment (alisey) [inline _codux/boards/services-section.board.css:72]: `width: 100%` and `align-content: center` are unnecessary.

File: _codux/boards/services-section.board.css [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+  display: flex;
+  justify-content: end;
+  padding-bottom: 24px;
+}
+.servicesCardIcon {
+  width: 32px;
```
Comment (alisey) [inline _codux/boards/services-section.board.css:78]: Images should have both width and height when possible. To minimize layout changes during page load.

File: _codux/boards/services-section.board.css [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.servicesButton {
+  height: 50px;
```
Comment (alisey) [inline _codux/boards/services-section.board.css:82]: It's better to specify vertical padding instead of giving the button fixed height, so the button will more nicely adapt to font size changes.

File: _codux/boards/services-section.board.css (hunk 21 above)
Comment (alisey) [reply _codux/boards/services-section.board.css:72]: The entire `.servicesCardIconContainer` is unnecessary. You could remove them and do this instead:

```css
.servicesCardIcon {
    display: block;
    width: 32px;
    height: 32px;
    margin: 0 0 24px auto;
}
```

File: _codux/boards/services-section.board.css (hunk 16 above)
Comment (alisey) [reply _codux/boards/services-section.board.css:50]: Basically, `.servicesCards` simplifies to:

```css
.servicesCards {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 20px;
    @media (width <= 768px) {
        grid-template-columns: 1fr;
    }
}
```

**Comment by alisey** [comment]
@RanBerkovich1 

I've opened a PR with all of the comments fixed: https://github.com/wixplosives/wireframe-snippets/pull/4

Here's the list of changes compared to this PR: https://github.com/wixplosives/wireframe-snippets/pull/4/commits/4f0890b5c61d3232da478905b198d239088413c2
